### PR TITLE
Added Jenkins integration files and binary compilation scripts

### PR DESCRIPTION
**Pull request by lreciomelero**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+@Library('libpipelines') _
+
+hose {
+    EMAIL = 'eos@stratio.com'
+    BUILDTOOL = 'make'
+    DEVTIMEOUT = 30
+    ANCHORE_POLICY = "production"
```
Comment (stg-0) [inline Jenkinsfile:7]: could we remove this and use the default?

File: .gitignore [hunk 2]
```diff
@@ -20,3 +20,8 @@
 *.sublime-project
 *.sublime-workspace
 *~
+
+deploy-recorder.lst
+/hack/go
+/hack/pkg
+/hack/*.tar.gz
```
Comment (stg-0) [inline .gitignore:27]:
```suggestion
/hack/*.tar.gz

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Changelog
+
+## 0.1.0 (upcoming)
+
+* Add clusterAPI for AWS capabilities
```
Comment (stg-0) [inline CHANGELOG.md:5]:
```suggestion
* Add clusterAPI capabilities for EKS
```

File: Makefile [hunk 4]
```diff
@@ -110,5 +110,12 @@ lint:
...
+
+deps:
+	hack/custom/deps.sh
+
+deploy: build
+	hack/custom/deploy.sh
+
 #################################################################################
-.PHONY: all kind build install unit clean update generate gofmt verify lint shellcheck
+.PHONY: all kind build install unit clean update generate gofmt verify lint shellcheck
```
Comment (stg-0) [inline Makefile:121]:
```suggestion
.PHONY: all kind build install unit clean update generate gofmt verify lint shellcheck

```

File: VERSION [hunk 5]
```diff
@@ -0,0 +1 @@
+0.1.0-SNAPSHOT
```
Comment (stg-0) [inline VERSION:1]:
```suggestion
0.1.0-SNAPSHOT

```

File: .gitignore (hunk 2 above)
Comment (lreciomelero) [reply .gitignore:27]: To ensure the version of go that is used, during the installation of the dependencies, I download the version that is required and then remove it. 
This line ensures that if an error occurs, the go binary is not pushed.

Comment (stg-0) [reply .gitignore:27]: I'm just adding the newline.. :)

File: hack/custom/deploy.sh [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+if [ -d "$DIR" ] || [ -r "$DIR"/cloud-provisioner]; then
+	echo "Uploading cloud-provisioner-$VERSION..."
+	tar czf "$DIR"/cloud-provisioner-${VERSION}.${EXTENSION} "$DIR"/cloud-provisioner
+	curl -sS -u stratio:${NEXUSPASS} --upload-file "$DIR"/cloud-provisioner-${VERSION}.${EXTENSION} http://qa.int.stratio.com/${GROUP_ID_NEXUS}/
+  echo "$GROUP_ID:cloud-provisioner:$EXTENSION" >> "$BASEDIR/deploy-recorder.lst"
+  rm -rf $BASEDIR/hack/go
+else
+	echo "Run 'make build' first"
+	exit 1
+fi
```
Comment (stg-0) [inline hack/custom/deploy.sh:20]:
```suggestion
fi

```

**Review by stg-0 — APPROVED** [review]

**Pull request merged**